core-java-modules/core-java-io-5/src/test/java/com/baeldung/charstreams/CharStreamsUnitTest.java

```diff
@@ -0,0 +1,116 @@
+package com.baeldung.charstreams;
+
+import static org.junit.jupiter.api.Assertions.*;
+
+import java.io.*;
+import java.nio.file.*;
+import java.nio.file.Paths;
+import java.time.LocalDateTime;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import org.junit.jupiter.api.*;
+
+public class CharStreamsUnitTest {
+
+    private static final DailyTodo TODO = new DailyTodo("Code", LocalDateTime.now(), true);
+    private static final DailyTodo TODO_1 = new DailyTodo("Sleep", LocalDateTime.now(), false);
+    private static final DailyTodo TODO_2 = new DailyTodo("Read", LocalDateTime.now(), false);
+    public static String TODO_LIST_PW_FILE = "src/test/resources/pw_todos.txt";
+    public static String TODO_LIST_FW_FILE = "src/test/resources/fw_todos.txt";
+    
+
+    @BeforeEach
+    public void createFiles() throws IOException {
+        Files.createDirectories(Paths.get(TODO_LIST_PW_FILE).getParent());
+        Files.createDirectories(Paths.get(TODO_LIST_FW_FILE).getParent());
+    }
+
+    @Test
+    public void whenUsingTextData_thenPrintWriterWritesToFile() throws IOException {
+        try {
+            pwFile = CharStreams.writingStringToFilePrintWriter(TODO_LIST_PW_FILE , TODO.toString());
+            assertThat(new File(TODO_LIST_PW_FILE)).hasContent(TODO.toString());
+        }catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+    }
+
+    @Test
+    public void whenUsingTextData_thenFileWriterWritesToFile() throws IOException {
+        try {
+            CharStreams.writingStringToFileFileWriter(TODO_LIST_FW_FILE, TODO.toString());
+            assertThat(new File(TODO_LIST_FW_FILE)).hasContent(TODO.toString());
+        } catch (FileNotFoundException fnfe) {
+            fnfe.printStackTrace();
+        }
+        
+    }

```

Comment:
For these two tests where we are using a String directly, I think it would be best to avoid using the TODO object and instead use a String directly. The reason for this change is that it can confuse the reader as they try to understand the purpose of the TODO object and why a String was not used directly.

Reply:
The TODO object is been converted into String. I use it to avoid hard coding the String every time I need it and to avoid creating an extra variable.

Reply:
I think this is going to be confusing for the reader. I can leave it as an outstanding point that we can get a 2nd opinion on from a senior editor when they review it.